src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1995,3 +1995,11 @@ function repeat_lengths!(longnew::AbstractVector, shortold::AbstractVector,
         counter += l
     end
 end
+
+# Disallowed operations that are a common mistake
+
+Base.getindex(::AbstractDataFrame, ::Symbol) =
+    throw(ArgumentError("syntax df[column] is not supported use df[!, column] instead"))
+
+Base.setindex!(::AbstractDataFrame, ::Any, ::Symbol) =
+    throw(ArgumentError("syntax df[column] is not supported use df[!, column] instead"))
```

Comment:
How about also accepting strings? They are as likely to be used as symbols.

Reply:
This is one of the things that is disputable. The time when `df[:col]` was accepted we did not accept strings for column indexing, so old code would be not affected by your proposal.

However, maybe pandas users might try this. So the error with `AbstractStrings` would be not for legacy code fixing but for new users. What do you think given this?

Reply:
Yes pandas and R users will likely try that. I'd say let's bite the bullet: if we add this, better cover all cases. We could also do the same for `Integer`.

Reply:
OK - changed. Let us see how it goes